client/src/components/modal-forms/NewPhaseDialog.js

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import { Button, Dialog } from '../generic';
+import { Box } from '@material-ui/core';
+import { makeStyles } from '@material-ui/core/styles';
+import { RenderTextField, RenderDateField } from '../fields';
+import { Field, Formik, Form as FormikForm } from 'formik';
+import { CreatePhaseSchema, ToastStatus } from '../../constants';
+import { createPhase } from '../../services/site';
+import { useToast } from '../../hooks';
+
+const useStyles = makeStyles(() => ({
+  formButton: {
+    maxWidth: '200px',
+  },
+  formRow: {
+    gap: '25px',
+  },
+}));
+
+export const NewPhaseDialog = ({ onSubmit, onClose, open }) => {
+  const { openToast } = useToast();
+  const classes = useStyles();
+  const initialValues = {
+    phaseName: '',
+    startDate: '',
+    endDate: '',
+  };
+
+  const handleCreatePhase = async (phase) => {
+    const phaseJson = {
+      name: phase.phaseName,
+      start_date: phase.startDate,
+      end_date: phase.endDate,
```

Comment:
I would let the service method handle the this transformation

Reply:
I was on the fence with this one! The reason that I didn't was that the service is more reusable if it expects the correct attribute names (ie: if another action also creates the phase, then I don't have to use wrong attribute names to call the function). The mapping I felt belonged to the form itself.

Reply:
But then I guess I put the logic in for fetchSites, but at least that one depends on the columns parameter that is also passed in to tell it how to format the data.